src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -571,13 +571,13 @@ private function parseDefinition($id, $service, $file, array $defaults)
      *
      * @throws InvalidArgumentException When errors occur
      *
-     * @return string|array A parsed callable
+     * @return string|array|Reference A parsed callable
      */
     private function parseCallable($callable, $parameter, $id, $file)
     {
         if (\is_string($callable)) {
             if ('' !== $callable && '@' === $callable[0]) {
-                throw new InvalidArgumentException(sprintf('The value of the "%s" option for the "%s" service must be the id of the service without the "@" prefix (replace "%s" with "%s").', $parameter, $id, $callable, substr($callable, 1)));
+                return $this->resolveServices($callable, $file);
```

Comment:
that's not specific enough - should be handled next to reading the "factory" attribute instead

Reply:
You mean adding a specific check for `@...` on [line 438](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php#L438)?

Reply:
I mean that while the change on the YamlLoader will allow `__invoke` on all callables (factories, but also eg configurators), the one on XmlLoader will only apply to factories. We should chose one side and be consistent with it :)

Reply:
I would vote for also adding it to the `XmlFileLoader` configurators - might as well make service referencing consistent across all service definition properties.

I'll make that line change now and push it up.